bluemira/magnetostatics/polyhedral_prism.py

```diff
@@ -0,0 +1,796 @@
+    current_direction:
+        Normalised current direction vector (3)
+    face_points:
+        Array of points on each face (n_face, n_points, 3)
+    face_normals:
+        Array of normalised normal vectors to the faces (pointing outwards)
+        (n_face, 3)
+    mid_points:
+        Array of face midpoints (n_face, 3)
+    n_sides:
+        Number of points in the face points (avoid reflected lists)
+    point:
+        Point at which to calculate the magnetic field (3)
+
+    Returns
+    -------
+    Magnetic field vector at the point (response to unit current density)
+    """
+    field = np.zeros(3)
+    for i in range(len(face_normals)):
+        normal = face_normals[i]
+        field += np.cross(current_direction, normal) * _surface_integral_fabbri(
+            face_points[i], normal, mid_points[i], n_sides[i], point
+        )
+    return MU_0_4PI * field
+
+
+class PolyhedralPrismCurrentSource(
+    PrismEndCapMixin, PolyhedralCrossSectionCurrentSource
+):
+    """
+    3-D polyhedral prism current source with a polyhedral cross-section and
+    uniform current distribution.
+
+    The current direction is along the local y coordinate.
+
+    The cross-section is specified in the local x-z plane.
+
+    Parameters
+    ----------
+    origin:
+        The origin of the current source in global coordinates [m]
+    ds:
+        The direction vector of the current source in global coordinates [m]
+    normal:
+        The normalised normal vector of the current source in global coordinates [m]
+    t_vec:
+        The normalised tangent vector of the current source in global coordinates [m]
+    xs_coordinates:
+        Coordinates of the conductor cross-section (specified in the x-z plane)
+    alpha:
+        The first angle of the trapezoidal prism [°] [0, 180)
+    beta:
+        The second angle of the trapezoidal prism [°] [0, 180)
+    current:
+        The current flowing through the source [A]
+
+    Notes
+    -----
+    Negative angles are allowed, but both angles must be equal
+    """
+
+    def __init__(
+        self,
+        origin: np.ndarray,
+        ds: np.ndarray,
+        normal: np.ndarray,
+        t_vec: np.ndarray,
+        xs_coordinates: Coordinates,
+        alpha: float,
+        beta: float,
+        current: float,
+    ):
+        alpha, beta = np.deg2rad(alpha), np.deg2rad(beta)
+        self._origin = origin
+
+        length = np.linalg.norm(ds)
+        self._halflength = 0.5 * length
+        self._check_angle_values(alpha, beta)
+        m_breadth = np.max(np.abs(xs_coordinates.x))
+        self._check_raise_self_intersection(length, m_breadth, alpha, beta)
+
+        # Normalised direction cosine matrix
+        self._dcm = np.array([t_vec, ds / length, normal])
+        self._set_cross_section(xs_coordinates)
+
+        self._alpha = alpha
+        self._beta = beta
+
+        # Current density
+        self.set_current(current)
+        self._points = self._calculate_points()
```

Comment:
just for my awareness where is this used? I can't find where it is.

Reply:
Yeah it is a necessary attribute coming from the baseclass (for plotting IIRC)